src/AppBundle/Controller/Oauth2Controller.php

```diff
@@ -108,83 +109,205 @@ public function loadDeckAction($id)
-		if(!$faction) {
+
+		// A deck ID was provided, so we lookup the deck that is being modified.
+		$deck = $this->getDoctrine()->getRepository('AppBundle:Deck')->find($id);
+		if($deck->getUser()->getId() !== $this->getUser()->getId()) {
+			throw $this->createAccessDeniedException("Access denied to this object.");
+		}
+		if ($deck->getNextDeck()) {
+			throw new BadRequestHttpException("Deck is locked");
+		}
+
+		// Don't allow investigator to be changed when 'editing' a deck.
+		// Seems unnecessary and is bound to break something.
+		$investigator = $deck->getCharacter();
+		if (!$investigator) {
 			return new JsonResponse([
-					'success' => FALSE,
-					'msg' => "Faction code invalid"
+				'success' => FALSE,
+				'msg' => "Investigator code invalid"
 			]);
 		}
-		
+
+		// Slots is the one required parameter.
 		$slots = (array) json_decode($request->get('slots'));
 		if (!count($slots)) {
 			return new JsonResponse([
 					'success' => FALSE,
 					'msg' => "Slots missing"
 			]);
 		}
-		foreach($slots as $card_code => $qty)
-		{
-			if(!is_string($card_code) || !is_integer($qty))
-			{
+
+		foreach($slots as $card_code => $qty) {
+			if(!is_string($card_code) || !is_integer($qty)) {
 				return new JsonResponse([
 						'success' => FALSE,
 						'msg' => "Slots invalid"
 				]);
 			}
 		}
-		
+		// We expect all
+		$problem = filter_var($request->get('problem'), FILTER_SANITIZE_STRING, FILTER_FLAG_NO_ENCODE_QUOTES);
+
 		$name = filter_var($request->get('name'), FILTER_SANITIZE_STRING, FILTER_FLAG_NO_ENCODE_QUOTES);
 		if(!$name) {
-			return new JsonResponse([
-					'success' => FALSE,
-					'msg' => "Name missing"
-			]);
+			if ($deck->getName()) {
+				$name = $deck->getName();
+			} else {
+				return new JsonResponse([
+						'success' => FALSE,
+						'msg' => "Name missing"
+				]);
+			}
 		}
-		
+
 		$decklist_id = filter_var($request->get('decklist_id'), FILTER_SANITIZE_NUMBER_INT);
+		if (!$decklist_id && $deck->getParent()) {
+			// Don't override the parent if this deck was copied and a param was not specified.
+			$decklist_id = $deck->getParent();
+		}
+
 		$description = trim($request->get('description'));
+		if (!$description && $deck->getDescriptionMd()) {
+			// Leave description alone if it was not specified (or was blank?).
+			$description = $deck->getDescriptionMd();
+		}
+
 		$tags = filter_var($request->get('tags'), FILTER_SANITIZE_STRING, FILTER_FLAG_NO_ENCODE_QUOTES);
-		
-		$this->get('decks')->saveDeck($this->getUser(), $deck, $decklist_id, $name, $faction, $description, $tags, $slots, null);
-		
-		$this->getDoctrine()->getManager()->flush();
-		
+		if (!$tags && $deck->getTags()) {
+			// Leave tags alone if they were blank.
+			$tags = $deck->getTags();
+		}
+
+		// Save the deck.
+		$this->get('decks')->saveDeck($this->getUser(), $deck, $decklist_id, $name, $investigator, $description, $tags, $slots, $deck , $problem);

```

Comment:
This is basically same as before, exception the faction -> investigator and we add support for previous deck (to support minor version) and problem (so deck list view still works).